specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json

```diff
@@ -5066,12 +5106,28 @@
           "format": "uuid",
           "readOnly": true,
           "description": "The client tenant id of the identity."
+        },
+        "userAssignedIdentities": {
+          "$ref": "#/definitions/UserAssignedIdentitiesDefinitionSpecification",
+          "description": "List of user assigned identities for the factory."
         }
       },
       "required": [
         "type"
       ]
     },
+    "UserAssignedIdentitiesDefinitionSpecification": {
+      "description": "Definition of all user assigned identities for a factory.",
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/UserAssignedIdentitySpecification"
+      }
+    },
+    "UserAssignedIdentitySpecification": {
```

Comment:
Why we have the two types "UserAssignedIdentitiesDefinitionSpecification" & "UserAssignedIdentitySpecification" and no explicit properties defined in them?

Reply:
It's a dictionary. I copied it from CosmosDB swagger
UserAssignedIdentities is an Object and the properties that it contains it's a Dictionar<string, Object> that's why we need both things. 

"UserAssignedIdentities": {
  "/subscriptions/b3ee3a7f-7614-4644-ad07-afa832620b4b/resourceGroups/ADF/providers/Microsoft.ManagedIdentity/userAssignedIdentities/PacodelTestMI": {},
  "/subscriptions/b3ee3a7f-7614-4644-ad07-afa832620b4b/resourceGroups/ADF/providers/Microsoft.ManagedIdentity/userAssignedIdentities/PacoTestMI2": {}
}